### PR TITLE
VACMS-13850: Creates `FormRouteSubscriber` service.

### LINKED ISSUE #13850: Create `ContentReleaseFormRouteSubscriber` service.
## Description

Currently we have a `\Drupal\va_gov_build_trigger\EventSubscriber\RouteSubscriber` event subscriber that sets the specific class of the form backing the build trigger form. This should be moved to a more specific class name, e.g. `ContentReleaseFormRouteSubscriber`, and use the `ContentReleaseFormResolver` service rather than the (current) environment discovery service. This should make it trivial to test.

## Acceptance Criteria

- [ ] The class has been created and uses the injected class to specify the form as expected.
- [ ] Unit tests cover this functionality.

### PR DESCRIPTION
**Pull request by ndouglas**
## Description

Closes #13850.

This does not yet replace the `va_gov_build_trigger` route subscriber; I have a separate issue for doing that, #14272.

### CODE REVIEW

File: docroot/modules/custom/va_gov_content_release/src/Form/Resolver/Resolver.php
Comment (ndouglas) [inline docroot/modules/custom/va_gov_content_release/src/Form/Resolver/Resolver.php:1]: Not relevant to the larger PR -- just a namespace change I shoehorned in.

File: docroot/modules/custom/va_gov_content_release/src/Strategy/Resolver/Resolver.php
Comment (ndouglas) [inline docroot/modules/custom/va_gov_content_release/src/Strategy/Resolver/Resolver.php:1]: Not relevant to the larger PR -- just a namespace change I shoehorned in.

File: docroot/modules/custom/va_gov_environment/src/Discovery/Discovery.php
Comment (ndouglas) [inline docroot/modules/custom/va_gov_environment/src/Discovery/Discovery.php:1]: Not relevant to the larger PR -- just a namespace change I shoehorned in.

File: docroot/modules/custom/va_gov_environment/src/Discovery/DiscoveryInterface.php
Comment (ndouglas) [inline docroot/modules/custom/va_gov_environment/src/Discovery/DiscoveryInterface.php:1]: Not relevant to the larger PR -- just a namespace change I shoehorned in.

File: docroot/modules/custom/va_gov_environment/va_gov_environment.services.yml
Comment (ndouglas) [inline docroot/modules/custom/va_gov_environment/va_gov_environment.services.yml:1]: Not relevant to the larger PR -- just a namespace change I shoehorned in.

File: tests/phpunit/va_gov_content_release/unit/Form/Resolver/ResolverTest.php
Comment (ndouglas) [inline tests/phpunit/va_gov_content_release/unit/Form/Resolver/ResolverTest.php:1]: Not relevant to the larger PR -- just a namespace change I shoehorned in.

File: tests/phpunit/va_gov_content_release/unit/Strategy/Resolver/ResolverTest.php
Comment (ndouglas) [inline tests/phpunit/va_gov_content_release/unit/Strategy/Resolver/ResolverTest.php:1]: Not relevant to the larger PR -- just a namespace change I shoehorned in.

File: tests/phpunit/va_gov_environment/functional/Service/DiscoveryTest.php
Comment (ndouglas) [inline tests/phpunit/va_gov_environment/functional/Service/DiscoveryTest.php:1]: Not relevant to the larger PR -- just a namespace change I shoehorned in.

File: tests/phpunit/va_gov_environment/unit/Environment/EnvironmentTest.php
Comment (ndouglas) [inline tests/phpunit/va_gov_environment/unit/Environment/EnvironmentTest.php:1]: Not relevant to the larger PR -- just a namespace change I shoehorned in.

File: tests/phpunit/va_gov_environment/unit/Service/DiscoveryTest.php
Comment (ndouglas) [inline tests/phpunit/va_gov_environment/unit/Service/DiscoveryTest.php:1]: Not relevant to the larger PR -- just a namespace change I shoehorned in.

**Review by edmund-dunn — APPROVED** [review]
Looks good! Clean code!

**Pull request merged**